atom/browser/atom_browser_client.cc

```diff
@@ -218,13 +218,14 @@ void AtomBrowserClient::SelectClientCertificate(
   auto cert_path = command_line->GetSwitchValueNative(
       switches::kClientCertificate);
 
-  // TODO(zcbenz): allow users to select certificate from
-  // client_cert list. Right now defaults to first certificate
-  // in the list.
   scoped_refptr<net::X509Certificate> certificate;
   if (cert_path.empty()) {
-    if (!cert_request_info->client_certs.empty())
-      certificate = cert_request_info->client_certs[0];
+    if (!cert_request_info->client_certs.empty()) {
+      auto window = NativeWindow::FromWebContents(web_contents);
+      if (window)
+        window->ClientCertificateSelector(cert_request_info,
```

Comment:
The `select-certificate` should be an event of `app` with `webContents` passed as parameter, because it happens for all WebContents instead of just for BrowserWindow, and under most cases users won't care the WebContents when selecting certificates.